Sofa/Component/Constraint/Lagrangian/Solver/src/sofa/component/constraint/lagrangian/solver/GenericConstraintSolver.cpp

```diff
@@ -1119,6 +1074,240 @@ void GenericConstraintProblem::unbuiltGaussSeidel(SReal timeout, GenericConstrai
+        {
+            convergence = true;
+            break;
+        }
+
+
+        // NNCG update with the correction p
+        for(int j=0; j<dimension; j++)
+        {
+            m_deltaF_new[j] = -(force[j] - m_lam[j]);
+        }
+
+        SReal beta = m_deltaF_new.dot(m_deltaF_new) / m_deltaF.dot(m_deltaF);
+        m_deltaF.eq(m_deltaF_new, 1);
+
+        if(beta > 1)
+        {
+            m_p.clear();
+            m_p.resize(dimension);
+        }
+        else
+        {
+            for(int j=0; j<dimension; j++)
+            {
+                force[j] += beta*m_p[j];
+                m_p[j] = beta*m_p[j] - m_deltaF[j];
+            }
+        }
+    }
+
+    result_output(force, error, iterCount, convergence);
+}
+
+void GenericConstraintProblem::gaussSeidel_increment(bool measureError, SReal *dfree, SReal *force, SReal **w, SReal tol, SReal *d, int dim, bool constraintsAreVerified, SReal& error, sofa::type::vector<SReal>& tabErrors)
+{
+    for(int j=0; j<dim; ) // increment of j realized at the end of the loop
+    {
+        //1. nbLines provide the dimension of the constraint
+        int nb = constraintsResolutions[j]->getNbLines();
+
+        //2. for each line we compute the actual value of d
+        //   (a)d is set to dfree
+
+        std::vector<SReal> errF(&force[j], &force[j+nb]);
+        std::copy_n(&dfree[j], nb, &d[j]);
+
+        //   (b) contribution of forces are added to d     => TODO => optimization (no computation when force= 0 !!)
+        for(int k=0; k<dim; k++)
+        {
+            for(int l=0; l<nb; l++)
+            {
+                d[j+l] += w[j+l][k] * force[k];
+            }
+        }
+
+        //3. the specific resolution of the constraint(s) is called
+        constraintsResolutions[j]->resolution(j, w, d, force, dfree);
+
+        //4. the error is measured (displacement due to the new resolution (i.e. due to the new force))
+        if(measureError)
+        {
+            SReal contraintError = 0.0;
+            if(nb > 1)
+            {
+                for(int l=0; l<nb; l++)
+                {
+                    SReal lineError = 0.0;
+                    for (int m=0; m<nb; m++)
+                    {
+                        SReal dofError = w[j+l][j+m] * (force[j+m] - errF[m]);
+                        lineError += dofError * dofError;
+                    }
+                    lineError = sqrt(lineError);
+                    if(lineError > tol)
+                    {
+                        constraintsAreVerified = false;
+                    }
+
+                    contraintError += lineError;
+                }
+            }
+            else
+            {
+                contraintError = fabs(w[j][j] * (force[j] - errF[0]));
+                if(contraintError > tol)
+                {
+                    constraintsAreVerified = false;
+                }
+            }
+
+            if(constraintsResolutions[j]->getTolerance())
```

Comment:
There is an implicit conversion to a `bool` here. Could you explicit your intentions?